sklearn/metrics/tests/test_score_objects.py

```diff
@@ -1356,3 +1371,20 @@ def score(y_true, y_pred, param=None):
             ValueError, match="is only supported if enable_metadata_routing=True"
         ):
             scorer(clf, X, y, param="blah")
+
+
+def test_get_scorer_multilabel_indicator():
+    """Check that our scorer deal with multi-label indicator matrices.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/26817
+    """
+    X, Y = make_multilabel_classification(n_samples=72, n_classes=3, random_state=0)
+    X_train, X_test, Y_train, Y_test = train_test_split(X, Y, random_state=0)
+
+    base_lr = LogisticRegression(solver="lbfgs", random_state=0)
+    chain = ClassifierChain(base_lr, order="random", random_state=0)
+    chain.fit(X_train, Y_train)
```

Comment:
We don't need to use the ClassifierChain here. Just using a KNeighborsClassifier reproduces the original issue.